### ISSUE TITLE
Theming Components

### ISSUE THREAD

**Issue by JayPanoz** [edrlab/thorium-web#58]
This is a placeholder issue to discuss theming + components. It’s not a priority right now and should be explored at a later date. 

Right now UI components are just applying the current’s theme properties when set by the user through CSS Properties (at the `:root` level). In the future, there could be an option to decouple UI from theming (either disable it entirely for UI Components, or rely on the OS preference (light/dark). Although I've heard about a11y in that context, I’ve not been able to verify it could be useful in that case. 

Considering we’re appending CSS Custom Properties through a custom hook (`useTheming`), it wouldn’t be too difficult to implement alternative options, although the devil is in the details, of course.

**Comment by JayPanoz** [comment]
Note to self: having shared styles (classnames) for all buttons works OK-ish for consistent sizing, etc. but may be an issue when it comes to theming (bg, color, fill, etc.). Maybe a better approach would be to scope them depending on their container (header/footer/sheet/etc.) but that obviously depends on the design of RS Preferences.
